services/web/server/src/simcore_service_webserver/groups_api.py

```diff
@@ -0,0 +1,310 @@
+import logging
+from typing import Dict, List, Optional, Tuple
+
+import sqlalchemy as sa
+from aiohttp import web
+from aiopg.sa import SAConnection
+from aiopg.sa.result import RowProxy
+from sqlalchemy import and_, literal_column
+
+from servicelib.application_keys import APP_DB_ENGINE_KEY
+
+from .db_models import GroupType, groups, user_to_groups, users
+from .groups_exceptions import (
+    GroupNotFoundError,
+    GroupsException,
+    UserInGroupNotFoundError,
+)
+from .groups_utils import (
+    check_group_permissions,
+    convert_groups_db_to_schema,
+    convert_groups_schema_to_db,
+    convert_user_in_group_to_schema,
+)
+from .users_exceptions import UserNotFoundError
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_GROUP_READ_ACCESS_RIGHTS = {"read": True, "write": False, "delete": False}
+DEFAULT_GROUP_OWNER_ACCESS_RIGHTS = {"read": True, "write": True, "delete": True}
+
+
```

Comment:
TIP: you can use alias in typing to document better your code and actually can be re-used in the future with pydantic's magic to convert between db and api models! For instance

```python

NameStr = str
PrimaryGroup = Dict[NameStr, str]
UserGroup = Dict[NameStr,str]

# ...

async def list_user_groups( ... ) -> Tuple[ PrimaryGroup, UserGroup, AllGroup]:
  # ...

```

Reply:
very cool idea!!!